rllib/core/models/tf/primitives.py

```diff
@@ -1,58 +1,62 @@
+import abc
 from typing import List
-from ray.rllib.utils.framework import try_import_tf
+
+from ray.rllib.core.models.base import Model, ModelConfig
+from ray.rllib.models.specs.checker import (
+    check_input_specs,
+    check_output_specs,
+)
 from ray.rllib.models.specs.checker import (
     is_input_decorated,
     is_output_decorated,
 )
-from ray.rllib.models.temp_spec_classes import TensorDict
-from ray.rllib.models.experimental.base import Model
-from ray.rllib.utils.typing import TensorType
 from ray.rllib.models.utils import get_activation_fn
-from typing import Tuple
-from ray.rllib.models.specs.checker import (
-    check_input_specs,
-    check_output_specs,
-)
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.nested_dict import NestedDict
 
 _, tf, _ = try_import_tf()
 
 
-def _call_not_decorated(input_or_output):
-    return (
-        f"forward not decorated with {input_or_output} specification. Decorate "
-        f"with @check_{input_or_output}_specs() to define a specification. See "
-        f"BaseModel for examples."
+def _raise_not_decorated_exception(input_or_output):
+    raise ValueError(
+        f"`TfModel.__call__()` not decorated with {input_or_output} specification. "
+        f"Decorate it with @check_{input_or_output}_specs() to define a specification."
     )
 
 
-class TfModel(Model, tf.Module):
-    """Base class for RLlib models.
+class TfModel(Model, tf.Module, abc.ABC):
```

Comment:
The PR is mostly about TorchModels and BaseModels, as the descriptions says.
This TF Model is not tested, but I wanted to take some of the torch changes already and get this a little up to speed. These changes make it so that it at least conforms to what we find in Torch.